### ISSUE TITLE
Suffer from over fitting during validation!! Will dropout help in reducing val. losses and over-fitting ?

### ISSUE THREAD

**Issue by rakehsaleem** [matterport/Mask_RCNN#1796]
Hello, I have been looking to find a way to reduce overfitting for my model. Let me tell you that my training losses are converging but my validation losses are overfitting a big time and I have a good dataset of 1100 images with some augmentations applied. Since my objects are really thin like cracks so I need to remove overfitting.
I know dropout can be helpful in reducing overfitting problem and I have looked at your previous model resources but found dropout only in fpn_classifier_graph as mentioned in above comments.

I am not sure will adding dropout to this will solve the issue or do I have to add it in the conv. block of ROI pooling. Similarly, do I have to add it in the RPN network and Resnet graph also? Any help would be really appreciated.
@waleedka

**Comment by ghost** [comment]
@rakehsaleem did you solve this?

**Comment by rakehsaleem** [comment]
@sohinimallick Thanks for reaching out,
I have done this project in the past (at least more than a year ago) and at that time, I only tried adding an RPN network and to be honest, the results were nonetheless the same and I don't see much of a difference in validation loss. I could say maybe it's because of the reason I was working on crack damages (object with very fine line and random structure). I would like to know if you achieve some good results by adding this? Please share your findings here so others can know anything critical.

Eventually, I tried multi-stage training and keep a check on my training set up to stop when a sufficient level or loss is achieved. Hope this helps and please ask me any question if you have! Thanks :)
